# Исследование надежности заемщиков


**Описание проекта**

Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.

Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

**Описание данных**

* `children` — количество детей в семье;
* `days_employed` — общий трудовой стаж в днях;
* `dob_years` — возраст клиента в годах;
* `education` — уровень образования клиента;
* `education_id` — идентификатор уровня образования;
* `family_status` — семейное положение;
* `family_status_id` — идентификатор семейного положения;
* `gender` — пол клиента;
* `income_type` — тип занятости;
* `debt` — имел ли задолженность по возврату кредитов;
* `total_income` — ежемесячный доход;
* `purpose` — цель получения кредита.

<h1>Содержание<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Изучение-общей-информации" data-toc-modified-id="Изучение-общей-информации-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Изучение общей информации</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Удаление-пропусков" data-toc-modified-id="Удаление-пропусков-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Удаление пропусков</a></span></li><li><span><a href="#Обработка-аномальных-значений" data-toc-modified-id="Обработка-аномальных-значений-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Обработка аномальных значений</a></span></li><li><span><a href="#Удаление-пропусков-(продолжение)" data-toc-modified-id="Удаление-пропусков-(продолжение)-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Удаление пропусков (продолжение)</a></span></li><li><span><a href="#Изменение-типов-данных" data-toc-modified-id="Изменение-типов-данных-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Изменение типов данных</a></span></li><li><span><a href="#Обработка-дубликатов" data-toc-modified-id="Обработка-дубликатов-2.5"><span class="toc-item-num">2.5&nbsp;&nbsp;</span>Обработка дубликатов</a></span></li><li><span><a href="#Категоризация-данных" data-toc-modified-id="Категоризация-данных-2.6"><span class="toc-item-num">2.6&nbsp;&nbsp;</span>Категоризация данных</a></span></li></ul></li><li><span><a href="#Исследование-данных-и-ответы-на-вопросы" data-toc-modified-id="Исследование-данных-и-ответы-на-вопросы-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Исследование данных и ответы на вопросы</a></span><ul class="toc-item"><li><span><a href="#Есть-ли-зависимость-между-количеством-детей-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-количеством-детей-и-возвратом-кредита-в-срок?-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Есть ли зависимость между количеством детей и возвратом кредита в срок?</a></span></li><li><span><a href="#Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Есть ли зависимость между семейным положением и возвратом кредита в срок?</a></span></li><li><span><a href="#Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Есть ли зависимость между уровнем дохода и возвратом кредита в срок?</a></span></li><li><span><a href="#Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?" data-toc-modified-id="Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Как разные цели кредита влияют на его возврат в срок?</a></span></li><li><span><a href="#Приведите-возможные-причины-появления-пропусков-в-исходных-данных." data-toc-modified-id="Приведите-возможные-причины-появления-пропусков-в-исходных-данных.-3.5"><span class="toc-item-num">3.5&nbsp;&nbsp;</span>Приведите возможные причины появления пропусков в исходных данных.</a></span></li><li><span><a href="#Объясните,-почему-заполнить-пропуски-медианным-значением-—-лучшее-решение-для-количественных-переменных." data-toc-modified-id="Объясните,-почему-заполнить-пропуски-медианным-значением-—-лучшее-решение-для-количественных-переменных.-3.6"><span class="toc-item-num">3.6&nbsp;&nbsp;</span>Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.</a></span></li></ul></li><li><span><a href="#Общий-вывод" data-toc-modified-id="Общий-вывод-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Общий вывод</a></span></li></ul></div>

In [6]:
import pandas as pd

## Изучение общей информации

In [7]:
try:
    data = pd.read_csv('datasets\data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

In [8]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

In [9]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

In [10]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

В столбцах `days_employed` и `total_income` есть пропущенные значения.

На сумму дохода сильнее всего влияет тип занятости, поэтому заполним пропуски в столбце `total_income` медианным значением по каждому типу из столбца `income_type`.

Пропуски в столбце `days_employed` обработаем позже.

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

In [12]:
data.describe()

children  days_employed     dob_years  education_id  \
count  21525.000000   19351.000000  21525.000000  21525.000000   
mean       0.538908   63046.497661     43.293380      0.817236   
std        1.381587  140827.311974     12.574584      0.548138   
min       -1.000000  -18388.949901      0.000000      0.000000   
25%        0.000000   -2747.423625     33.000000      1.000000   
50%        0.000000   -1203.369529     42.000000      1.000000   
75%        1.000000    -291.095954     53.000000      1.000000   
max       20.000000  401755.400475     75.000000      4.000000   

       family_status_id          debt  total_income  
count      21525.000000  21525.000000  2.152500e+04  
mean           0.972544      0.080883  1.652253e+05  
std            1.420324      0.272661  9.804367e+04  
min            0.000000      0.000000  2.066726e+04  
25%            0.000000      0.000000  1.077982e+05  
50%            0.000000      0.000000  1.425944e+05  
75%            1.000000      0.000000  1.955499e+05  
max            4.000000      1.000000  2.265604e+06

В столбце `days_employed` имеется отрицательное количество дней трудового стажа. 

Заменим все отрицательные значения положительными.

In [13]:
data['days_employed'] = data['days_employed'].abs()

Выведем для каждого типа занятости медианное значение трудового стажа в днях.

In [14]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) аномально большие значения. 

Поскольку столбец не понадобится для исследования, оставим значения как есть.

In [15]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

В столбце `children` есть два аномальных значения. 

Удалим строки, в которых встречаются такие аномальные значения.

In [16]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

In [17]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

### Удаление пропусков (продолжение)

Заполним пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.

In [18]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

In [19]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

Заменим вещественный тип данных в столбце `total_income` на целочисленный.

In [20]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

In [21]:
data['education'].unique()

array(['высшее', 'среднее', 'Среднее', 'СРЕДНЕЕ', 'ВЫСШЕЕ',
       'неоконченное высшее', 'начальное', 'Высшее',
       'НЕОКОНЧЕННОЕ ВЫСШЕЕ', 'Неоконченное высшее', 'НАЧАЛЬНОЕ',
       'Начальное', 'Ученая степень', 'УЧЕНАЯ СТЕПЕНЬ', 'ученая степень'],
      dtype=object)

В столбце `education` имеются неявные дубликаты. 

Обработаем их, приведя к нижнему регистру.

In [22]:
data['education'] = data['education'].str.lower()

Проверим наличие неявных дубликатов в остальных столбцах.

In [23]:
for col in ['family_status', 'gender', 'income_type', 'purpose']:
    print(f'{col}:')
    display(data[col].unique())

family_status:


array(['женат / замужем', 'гражданский брак', 'вдовец / вдова',
       'в разводе', 'Не женат / не замужем'], dtype=object)

gender:


array(['F', 'M', 'XNA'], dtype=object)

income_type:


array(['сотрудник', 'пенсионер', 'компаньон', 'госслужащий',
       'безработный', 'предприниматель', 'студент', 'в декрете'],
      dtype=object)

purpose:


array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

В столбце `purpose` имеются неявные дубликаты. Обработаем их чуть позже.

Проверим существование явных дубликатов.

In [24]:
data.duplicated().sum()

71

Удалим дубликаты.

In [25]:
data = data.drop_duplicates()

### Категоризация данных

Создадим в датафрейме столбец `total_income_category` с категориями на основе следующих диапазонов:
- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


Для этого создадим функцию `categorize_income()` и применим ее к столбцу `total_income`.

In [26]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [27]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

In [28]:
data['total_income_category'].isna().sum()

0

На основании данных из столбца `purpose` сформируем новый столбец `purpose_category`, в который войдут следующие категории:
- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

Для этого создадим функцию `categorize_purpose()` и применим ее к столбцу `purpose`.

In [29]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [30]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

## Исследование данных и ответы на вопросы

### Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [31]:
pd.DataFrame(data.groupby('children')['debt'].mean() * 100).reset_index()

children      debt
0         0  7.543822
1         1  9.234609
2         2  9.454191
3         3  8.181818
4         4  9.756098
5         5  0.000000

In [32]:
data[data['children'] == 5]

children  days_employed  dob_years  education  education_id  \
3979          5    1572.328285         42    среднее             1   
4397          5    3248.839837         36    среднее             1   
7866          5     773.124856         36    среднее             1   
15822         5     418.199982         31    среднее             1   
15916         5    2286.262752         37    среднее             1   
16211         5     387.317579         35    среднее             1   
20452         5     268.425464         38  начальное             3   
20837         5    2386.600221         35    среднее             1   
21156         5    1690.018117         59    среднее             1   

          family_status  family_status_id gender  income_type  debt  \
3979   гражданский брак                 1      M    сотрудник     0   
4397    женат / замужем                 0      F    компаньон     0   
7866    женат / замужем                 0      F    сотрудник     0   
15822   женат / замужем                 0      F    сотрудник     0   
15916   женат / замужем                 0      F    сотрудник     0   
16211  гражданский брак                 1      F  госслужащий     0   
20452   женат / замужем                 0      F    сотрудник     0   
20837   женат / замужем                 0      F    компаньон     0   
21156   женат / замужем                 0      M    сотрудник     0   

       total_income                           purpose total_income_category  \
3979         142594      на покупку своего автомобиля                     C   
4397         168460          операции с недвижимостью                     C   
7866          48772                 операции с жильем                     D   
15822         77552  сделка с подержанным автомобилем                     C   
15916        256698              покупка недвижимости                     B   
16211        126102             на проведение свадьбы                     C   
20452        212545      заняться высшим образованием                     B   
20837        204241                             жилье                     B   
21156        269068   операции со своей недвижимостью                     B   

               purpose_category  
3979     операции с автомобилем  
4397   операции с недвижимостью  
7866   операции с недвижимостью  
15822    операции с автомобилем  
15916  операции с недвижимостью  
16211        проведение свадьбы  
20452     получение образования  
20837  операции с недвижимостью  
21156  операции с недвижимостью

**Вывод** 

Кредитополучатели без детей являются более ответственными, чем клиенты, у которых есть дети. Видимо, наличие ребенка накладывает дополнительное финансовое бремя, и клиенты либо менее ответственно относятся к своим кредитным обязательствам, либо не имеют возможности гасить кредит в срок.

Также нет должников среди клиентов, у которых пятеро детей. Однако из-за того, что во всем наборе данных таких клиентов всего девять, утверждать, что клиенты с пятью детьми в семье так же хорошо платят по кредитам, как и клиенты без детей, нельзя.

### Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [33]:
fam_stat_group = pd.DataFrame(data.groupby('family_status')['debt'].mean() * 100).reset_index()
fam_stat_group

family_status      debt
0  Не женат / не замужем  9.763948
1              в разводе  7.064760
2         вдовец / вдова  6.624606
3       гражданский брак  9.313014
4        женат / замужем  7.560558

Укрупним группы до `'Были (сейчас) женаты официально'` и `'Не были женаты официально'`.

In [34]:
family_status_group = {
    'Были (сейчас) женаты официально': ['в разводе', 'вдовец / вдова', 'женат / замужем'],
    'Не были женаты официально': ['Не женат / не замужем', 'гражданский брак']
}

debt = []

for status, statuses in family_status_group.items():
    mean_debt = fam_stat_group[fam_stat_group['family_status'].isin(statuses)]['debt'].mean()
    debt.append((status, mean_debt))
    
pd.DataFrame(debt, columns=['family_status_group', 'debt'])

family_status_group      debt
0  Были (сейчас) женаты официально  7.083308
1        Не были женаты официально  9.538481

**Вывод** 

Менее ответственными являются клиенты, которые не состоят или не состояли в браке. Узаконенные отношения (включая и бывшие отношения) ведут к более ответственному отношению к своим обязательствам.

### Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [35]:
pd.DataFrame(data.groupby('total_income_category')['debt'].mean() * 100).reset_index()

total_income_category      debt
0                     A  8.000000
1                     B  7.060231
2                     C  8.498210
3                     D  6.017192
4                     E  9.090909

**Вывод** 

Выборка по заемщикам, как с наименьшим доходом, так и с наибольшим, крайне недостаточна для формирования однозначных выводов. Можно думать, что самая низкодоходная E-категория не может корректно распоряжаться деньгами и "влезает" в кредиты, но неясно почему у А-категории также достаточно высокий процент задолженности.

Если опираться только на группы В, С и D, то по полученным данным не наблюдается зависимость "те, кто получает больше ЗП, лучше возвращает кредит".

### Как разные цели кредита влияют на его возврат в срок?

In [36]:
pd.DataFrame(data.groupby('purpose_category')['debt'].mean() * 100).reset_index()

purpose_category      debt
0    операции с автомобилем  9.347978
1  операции с недвижимостью  7.255139
2     получение образования  9.252758
3        проведение свадьбы  7.911803

**Вывод** 

Чаще в срок возвращаются кредиты, взятые на операции с недвижимостью и на проведение свадьбы.

### Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Отфильтровав данные следующей командой до этапа заполнения пропущенных значений можно увидеть, что все пропуски присутствуют в записях одновременно. Возможной причиной этого является техническая ошибка при формировании таблицы с исходными данными (либо некий сбой во время выгрузки данных на анализ; либо при объединении данных в единый файл, который также должен был передаваться на анализ; либо эти данные не собирались).


In [5]:
data[(data['days_employed'].isna()) & (data['total_income'].isna())]

children  days_employed  dob_years education  education_id  \
12            0            NaN         65   среднее             1   
26            0            NaN         41   среднее             1   
29            0            NaN         63   среднее             1   
41            0            NaN         50   среднее             1   
55            0            NaN         54   среднее             1   
...         ...            ...        ...       ...           ...   
21489         2            NaN         47   Среднее             1   
21495         1            NaN         50   среднее             1   
21497         0            NaN         48    ВЫСШЕЕ             0   
21502         1            NaN         42   среднее             1   
21510         2            NaN         28   среднее             1   

               family_status  family_status_id gender  income_type  debt  \
12          гражданский брак                 1      M    пенсионер     0   
26           женат / замужем                 0      M  госслужащий     0   
29     Не женат / не замужем                 4      F    пенсионер     0   
41           женат / замужем                 0      F  госслужащий     0   
55          гражданский брак                 1      F    пенсионер     1   
...                      ...               ...    ...          ...   ...   
21489        женат / замужем                 0      M    компаньон     0   
21495       гражданский брак                 1      F    сотрудник     0   
21497        женат / замужем                 0      F    компаньон     0   
21502        женат / замужем                 0      F    сотрудник     0   
21510        женат / замужем                 0      F    сотрудник     0   

       total_income                           purpose  
12              NaN                   сыграть свадьбу  
26              NaN                       образование  
29              NaN  строительство жилой недвижимости  
41              NaN  сделка с подержанным автомобилем  
55              NaN                   сыграть свадьбу  
...             ...                               ...  
21489           NaN              сделка с автомобилем  
21495           NaN                           свадьба  
21497           NaN        строительство недвижимости  
21502           NaN  строительство жилой недвижимости  
21510           NaN           приобретение автомобиля  

[2174 rows x 12 columns]

### Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* В данном проекте пропуски присутствовали в данных о стаже работы и о зарплате. Такие данные в различных ситуациях могут сильно разниться, например, в случаях работающих и неработающих студентов и пенсионеров. Поскольку медиана является более устойчивым к выбросам в данных показателем, ее использование является лучшим решением.

## Общий вывод

Были выполнены проверки зависимости возврата кредита от различных факторов. Входные данные — статистика о платёжеспособности клиентов. Результаты исследования будут учтены при построении модели кредитного скоринга — системы, оцениваающей способность потенциального заёмщика вернуть кредит банку.

Выводы по проверкам:
- Есть ли зависимость между количеством детей и возвратом кредита в срок?

Отсутствие детей в семье повышает возможность возврата кредита в срок (доля просрочек - 7.54%). У заемщиков с детьми доля просрочек варьируется в пределах 8.18-9.76%. При этом у заемщиков с пятью детьми просрочек нет вообще, но для выводов по данной категории необходимо провести дополнительное исследование.
- Есть ли зависимость между семейным положением и возвратом кредита в срок? 

Клиенты, которые были или сейчас женаты официально, имеют наименьший риск просрочки (доля просрочек - 7.08%). У заемщиков, которыке не были женаты официально, коэффициент просрочки составляет 9.54%.
- Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Однозначный вывод по имеющимся данным сделать нельзя. Результаты анализа показывают, что самая низкодоходная E (от 0 до 30.000 у.е.) и самая высокодоходная A (от 1.000.001 у.е. и выше) категории клиентов примерно однинаово просрачивают возврат кредита (9.09% и 8% просрочек соответственно).

Если опираться только на группы В, С и D (коэффициенты просрочки 7.06%, 8.5%, 6.02% при доходе 200.001–1.000.000 у.е., 50.001–200.000 у.е. и 30.001–50.000 у.е. соответственно), то по полученным данным не наблюдается зависимость "те, кто получает больше ЗП, лучше возвращает кредит".

- По цели кредита

Заемщики чаще возвращают в срок кредиты, взятые на операции с недвижимостью и на проведение свадьбы (доля просрочек - 7.3% и 7.9% соответственно). Доля просрочек по кредитам, выданным на получение образования и на операции с автомобилем составляет 9.26% и 9.35% соответственно.